### PR TITLE
Rmd refactor4 pd

### PR DESCRIPTION
**Pull request by pdiakumis**
@skanwal sorry for delay, I've done the refactor of the gene summaries as discussed. I think there was a bug in master at https://github.com/umccr/RNAsum/blob/e1dc8ebe1dea0f7ea310eb5337340edcfb0110c6/rmd_files/RNAseq_report.Rmd#L2407 (instead of 'Mutated', that should be 'Fusion').
Ignore the auto-generated `man/` files, main changes are in the Rmd and `R/sample_data.R`.

### CODE REVIEW

**Comment by skanwal** [comment]
Thanks @pdiakumis.

> I think there was a bug in master

You're right. I had picked it up during refactor work and corrected it here https://github.com/umccr/RNAsum/blob/rmd_refactor4/inst/rmd/rnasum.Rmd#L472 in my  refactor branch.

**Comment by skanwal** [comment]
I have tested this PR and the changes work nicely. 
I am thinking we can go through these quickly tomorrow in our 1-1 and then merge it in `rmd_refactor4` branch.

**Comment by pdiakumis** [comment]
@skanwal I refactored a couple more chunks, nothing too fancy.
I've also added some lib size barplot questions we can talk about next week.

File: inst/rmd/rnasum.Rmd [hunk 1]
```diff
@@ -425,157 +421,99 @@ tmpg <- tmpg |>
...
+genes2keep <- gene_info |>
+  dplyr::filter(.data$SYMBOL %in% genes2keep)
 genes2keep <- genes2keep[!duplicated(genes2keep$ENSEMBL), ]
 rownames(genes2keep) <- genes2keep$ENSEMBL
 
 ##### Remove rows with duplicated gene symbols (Y_RNAs, SNORs, LINC0s etc). Preferably select ENSEMBL ID that is used in the count data
-genes2keep.combined_data <- genes2keep[genes2keep$ENSEMBL %in% rownames(ref_dataset.list[[dataset]]$combined_data), ]
-genes2keep <- genes2keep[genes2keep$SYMBOL %!in% genes2keep.combined_data$SYMBOL, ]
+genes2keep.combined_data <- genes2keep |>
+  dplyr::filter(.data$ENSEMBL %in% rownames(ref_dataset.list[[dataset]][["combined_data"]]))
```
Comment (skanwal) [inline inst/rmd/rnasum.Rmd:514]: are we going to keep `rownames` call as last time you mentioned this is error prone?

**Review by skanwal — APPROVED** [review]

**Pull request merged**